### PR TITLE
Search suggestions: don't jump to result for different filter type

### PR DESCRIPTION
**Pull request by olafurpg**
Previously, accepting the top suggestion with Enter for the query `type:commit file:insights repo:sour<TAB>` would jump straight to the Sourcegraph repo and discard the fact that the query has `type:commit`. Now, pressing enter on repo/file/symbol suggestions doesn't jump to the result unless the query has a matching type.

Fixes https://sourcegraph.slack.com/archives/C03CSAER9LK/p1673524313583519

## Test plan

* Write query `type:commit repo:sourcegraph<TAB>`
* Press enter to accept the top suggestion
* Confirm that it updates the query instead of navigating to the URL of the sourcegraph repo.

## App preview:

- [Web](https://sg-web-olafurpg-suggestions-filter.onrender.com/search)

Check out the [client app preview documentation](https://docs.sourcegraph.com/dev/how-to/client_pr_previews) to learn more.

### CODE REVIEW

**Comment by sg-e2e-regression-test-bob** [comment]
## Bundle size report 📦

| Initial size | Total size | Async size | Modules |
| --- | --- | --- | --- |
| 0.00% (0.00 kb) | 0.00% (+0.16 kb) | 0.00% (+0.16 kb) | 0.00% (0) |

Look at the [Statoscope report](https://console.cloud.google.com/storage/browser/_details/sourcegraph_reports/statoscope-reports/olafurpg/suggestions-filter/compare-report.html;tab=live_object) for a full comparison between the commits 295b1ea and 1c8d4ab or [learn more](https://docs.sourcegraph.com/dev/how-to/testing#bundlesize).


<details>
<summary>Open explanation</summary>

- `Initial size` is the size of the initial bundle (the one that is loaded when you open the page)
- `Total size` is the size of the initial bundle + all the async loaded chunks
- `Async size` is the size of all the async loaded chunks
- `Modules` is the number of modules in the initial bundle
</details>

File: client/branded/src/search-ui/input/codemirror/completion.ts [hunk 1]
```diff
@@ -551,6 +551,23 @@ export function suggestionTypeFromTokens(
 
 type CompletionWithURL = Completion & { url?: string }
 
+// Returns the value of `type:VALUE` if it exists, or undefined otherwise.
+function queryType(tokens: Token[]): string | undefined {
+    for (const token of tokens) {
+        if (token.type === 'filter' && token.field.value === 'type') {
```
Comment (leonore) [inline client/branded/src/search-ui/input/codemirror/completion.ts:557]: I'm assuming this corresponds to the `type:file` search filters, do we want to apply this new behaviour to `select` filters as well?

**Review by leonore — COMMENTED** [review]
thanks you @olafurpg! the logic makes sense to me but I'd get a frontend engineer to review this as I don't want to overlook anything that this new behaviour might change!

**Comment by olafurpg** [comment]
Closing this PR since I don't think the change is 100% solid and I'm not planning to complete the remaining work.

**Pull request closed without merging**